sdk/servicebus/Microsoft.Azure.ServiceBus/tests/MessageReceiverTests.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.ServiceBus.UnitTests
+{
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.ServiceBus.Core;
+    using Xunit;
+
+    public sealed class MessageReceiverTests : SenderReceiverClientTestBase
```

Comment:
>MessageReceiverTests [](start = 24, length = 20)

We don't have a test which ensures that once `Unregister` is called, there are no more messages that are received, i.e, ensure no handlers are called after that